src/Components/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { DropdownProps, DropdownOption } from "./types";
+import { Autocomplete, TextField, Box, InputAdornment } from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search"; // Import the search icon
+import { DropDownType } from "./consts";
+
+const DropDown = ({
+  type = DropDownType.REGULAR,
+  options,
+  value,
+  onChange,
+  label = "",
+  style,
+}: DropdownProps) => {
+  const [showSearchIcon, setShowSearchIcon] = useState(
+    type == DropDownType.SEARCH
+  );
+  const handleChange = (event: any, newValue: DropdownOption | null) => {
+    setShowSearchIcon(false);
+    if (onChange && newValue) {
+      onChange(newValue.value);
+    } else if (!newValue) {
+      onChange("");
+      setShowSearchIcon(type == DropDownType.SEARCH);
+    }
+  };
+
+  return (
+    <Autocomplete
+      options={options}
+      getOptionLabel={(option) => option.label}
+      value={options.find((option) => option.value === value) || null}
+      onChange={handleChange}
+      renderInput={(params) =>
+        !showSearchIcon ? (
+          <TextField {...params} label={label} sx={style} variant="outlined" />
+        ) : (
+          <TextField
+            {...params}
+            placeholder={label}
+            sx={style}
+            variant="outlined"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <InputAdornment position="start">
+                  <SearchIcon style={{ cursor: "pointer" }} />
+                </InputAdornment>
+              ),
+            }}
+          />
+        )
+      }
```

Comment:
export this function to a util function ( in a new file named util.ts).
import it in this file and use it like : 
`renderInput={renderInput}`